prov/gni/src/gnix_nic.c

```diff
@@ -1253,6 +1253,10 @@ int gnix_nic_alloc(struct gnix_fid_domain *domain,
 		GNIX_INFO(FI_LOG_EP_CTRL, "Allocated NIC:%p\n", nic);
 	}
 
+	if (nic) {
+		nic->requires_lock = domain->thread_model != FI_THREAD_COMPLETION;
+	}
```

Comment:
i was referring to the way we allocate ```gnix_nic```s as ep's are created.  I think for it to be safe to set ```nic->requires_lock``` to be set to 0, we'd need to insure that each EP was associated with a unique ```gnix_nic``` instance, or alternatively, make a change such that each ```gnix_cq``` for a domain with FI_THREAD_COMPLETION, is associated with a single CQ.  But I agree that can be done in a subsequent PR.